x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/telemetry/InferenceAPMStats.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.telemetry;
+
+import org.elasticsearch.inference.Model;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.Objects;
+
+public class InferenceAPMStats extends InferenceStats {
+
+    private final LongCounter inferenceAPMRequestCounter;
+
+    public InferenceAPMStats(Model model, MeterRegistry meterRegistry) {
+        super(model);
+        this.inferenceAPMRequestCounter = meterRegistry.registerLongCounter(
```

Comment:
Another potential option is to use an async counter. From the docs it seems like a long counter is preferred when trying to have a monotonously increasing counter: https://github.com/elastic/elasticsearch/blob/main/modules/apm/METERING.md#how-to-choose-an-instrument

If we used an async counter we could pass in a callback and have it reference an AtomicLong within the `InferenceStats` class (currently it's using a `LongAdder`). Maybe that'd save some memory and complexity with having to increment two values here? 🤷‍♂️